src/vm/arm64/stubs.cpp

```diff
@@ -1307,28 +1314,52 @@ LONG CLRNoCatchHandler(EXCEPTION_POINTERS* pExceptionInfo, PVOID pv)
     return EXCEPTION_CONTINUE_SEARCH;
 }
 
+#ifndef CROSSGEN_COMPILE
 void StompWriteBarrierEphemeral(bool isRuntimeSuspended)
 {
-    return;
+    _ASSERTE(isRuntimeSuspended);
+
+    JIT_UpdateWriteBarrierState(GCHeapUtilities::IsServerHeap());
 }
 
 void StompWriteBarrierResize(bool isRuntimeSuspended, bool bReqUpperBoundsCheck)
 {
-    return;
+    // The runtime is not always suspended when this is called.  If a thread was
+    // preempted while holding the old card_table ptr in a register, when
+    // resumes it will write to the old card_table (bad).  Same is true for
+    // card_bundle_table
+    //
+    // Suspend the runtime to avoid this race condition
```

Comment:
> If a thread was preempted while holding the old card_table ptr in a register, when resumes it will write to the old card_table (bad)

This particular situation is not a problem. The old card table is still around when a bigger one gets allocated. The GC will clear the cardtables only when the threads are suspended.